cura/Settings/ContainerManager.py

```diff
@@ -694,6 +695,67 @@ def duplicateMaterial(self, material_id: str) -> str:
         duplicated_container.setDirty(True)
         self._container_registry.addContainer(duplicated_container)
 
+    #  Create a new material by cloning Generic PLA and setting the GUID to something unqiue
+    @pyqtSlot(result = str)
+    def createMaterial(self) -> str:
+        # Ensure all settings are saved.
+        Application.getInstance().saveSettings()
+
+        containers = self._container_registry.findInstanceContainers(id="generic_pla")
+        if not containers:
+            Logger.log("d", "Unable to create a new material by cloning generic_pla, because it doesn't exist.")
+            return ""
+
+        # Create a new ID & container to hold the data.
+        new_id = self._container_registry.uniqueName("custom_material")
+        container_type = type(containers[0])  # Could be either a XMLMaterialProfile or a InstanceContainer
+        duplicated_container = container_type(new_id)
+
+        # Instead of duplicating we load the data from the basefile again.
+        # This ensures that the inheritance goes well and all "cut up" subclasses of the xmlMaterial profile
+        # are also correctly created.
+        with open(containers[0].getPath(), encoding="utf-8") as f:
+            duplicated_container.deserialize(f.read())
+
+        duplicated_container.setMetaDataEntry("base_file", new_id)
+        duplicated_container.setMetaDataEntry("GUID", str(uuid.uuid4()))
+        duplicated_container.setName(catalog.i18nc("@label", "Custom Material"))
+
+        self._container_registry.addContainer(duplicated_container)
```

Comment:
I think you forgot to return something here. What it returns is also not documented.

Edit: Further down you check against this return value by comparing it with `""`. This currently seems to work because when a material is successfully created you get `None` which is not equal to the empty string. But still, please change it.